packages/store-api/src/platforms/vtex/loaders/simulation.ts

```diff
@@ -0,0 +1,42 @@
+import DataLoader from 'dataloader'
+
+import type {
+  PayloadItem,
+  Simulation,
+} from '../clients/commerce/types/Simulation'
+import type { Options } from '..'
+import type { Clients } from '../clients'
+
+export const getSimulationLoader = (_: Options, clients: Clients) =>
+  new DataLoader<PayloadItem[], Simulation>(
+    async (items) => {
+      const simulated = await clients.commerce.checkout.simulation({
+        items: [...items.flat()],
+      })
+
+      const itemsIndices = items.reduce(
+        (acc, curr) => [...acc, curr.length + acc[acc.length - 1]],
```

Comment:
Seeing the PayloadItem interface, it doesn't have any property called length. Can you explain why you access curr.length?